src/main/scala/viper/silver/plugin/standard/inline/InlineRewrite.scala

```diff
@@ -51,6 +59,65 @@ trait InlineRewrite extends PredicateExpansion {
       .execute[Exp](expr)
   }
 
+  /**
+    * Given an inhale statement, look within it for any predicate accesses. If found, return the same inhale statement
+    * with the expanded value of the predicate.
+    *
+    * @param stmt The statement containing the inhale statement we wish to expana.
+    * @param method The method containing the inhale statement we wish to expand.
+    * @param program The Viper program for which we perform this expansion.
+    * @param cond The condition a predicate must satisfy to be expanded within an inhale statement.
+    * @return The expanded inhale statement.
+    */
+  private[this] def expandInhaleStatement(stmt: Stmt, method: Method, program: Program, cond: String => Boolean): Stmt =
+    ViperStrategy.Context[Set[String]]({
+      case tup @ (Inhale(expr), ctxt) =>
+        val expandedExpr = expandInhaleContainedExpr(expr, method, program, cond)
+        tup match {
+          case (inhaleStmt: Inhale, _) =>
+            val expandedInhaleStmt =
+              inhaleStmt.copy(exp = expandedExpr)(pos = inhaleStmt.pos, info = inhaleStmt.info, errT = inhaleStmt.errT)
+            (expandedInhaleStmt, ctxt)
+          case _ =>
+            // This is required to silence the compiler warning for non-exhaustive pattern matching
+            throw new Error("This error should never be thrown")
+        }
+    }, method.scopedDecls.map(_.name).toSet, Traverse.TopDown).execute[Stmt](stmt)
+
+  /**
+    * Given a predicate access or a predicate access predicate in the context of an inhale statement,
+    * expand the expression and produce it as a value.
+    *
+    * @param expr The expression (likely a PredicateAccess/PredicateAccessPredicate) that we want to expand.
+    * @param method The method in which this expansion occurs.
+    * @param program The Viper program for which we perform this expansion.
+    * @param cond The condition a predicate must satisfy to be expanded within an inhale statement.
+    * @return The expanded expression.
+    */
+  private[this] def expandInhaleContainedExpr(expr: Exp, method: Method, program: Program, cond: String => Boolean): Exp  =
+    ViperStrategy.Context[Set[String]]({
+      case exp @ (PredicateAccess(_, name), ctxt) =>
+        val isInUnfolding = isWithinUnfolding(ctxt)
+        if (cond(name) && !isInUnfolding) {
+          val maybePredBody = exp match {
+            case (pred: PredicateAccess, _) => pred.predicateBody(program, ctxt.c)
+            case _ =>
+              // This is required to silence the compiler warning for non-exhaustive pattern matching
+              throw new Error("This error should never be thrown")
+          }
+          (maybePredBody.get, ctxt)
+        } else exp
```

Comment:
I don't think `PredicateAccess`es can ever appear alone outside of a `PredicateAccessPredicate`. If they can, then `expandPredicates` would need to be fixed. If they can't, though, then `expandPredicates` should probably just be called directly.

Reply:
```js
method client(a: Ref, b: Ref) {
  inhale allowAccess(a, b)
  unfold allowAccess(a, b)

  a.f := 1
  b.f := 3

  assert b.f == 3
}
```
this is valid and it verifies though

Reply:
`allowAccess(a, b)` is sugar for `acc(allowAccess(a, b), 1)`
From the [tutorial](http://viper.ethz.ch/tutorial/?page=1&section=#predicates):
> The simple syntax P(...) has the same meaning as acc(P(...)), which in turn has the same meaning as acc(P(...), write).